group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupConfig.java

```diff
@@ -238,4 +264,39 @@ public int shareHeartbeatIntervalMs() {
     public int shareRecordLockDurationMs() {
         return shareRecordLockDurationMs;
     }
+
+    /**
+     * The share group auto offset reset strategy.
+     */
+    public ShareGroupAutoOffsetReset shareAutoOffsetReset() {
+        return ShareGroupAutoOffsetReset.forStrategy(shareAutoOffsetReset);
+    }
+
+    public enum ShareGroupAutoOffsetReset {
+        LATEST("latest"),
+        EARLIEST("earliest"),
+        UNKNOWN("unknown");
+
+        private final String strategy;
+
+        ShareGroupAutoOffsetReset(String strategy) {
```

Comment:
You certainly could use `OffsetResetStrategy` as a model for the enum. You don't need to use the string like this. Look at ConsumerConfig.java, where it uses `Utils.enumOptions` to initialize the value.